### PR TITLE
Pipeline config & schedule modeless bugfixes/improvements

### PR DESCRIPTION
**Pull request by tonybach**
JIRAs:

- [CDAP-10957](https://issues.cask.co/browse/CDAP-10957): Allows users to set max concurrent runs on batch pipelines
- [CDAP-11421](https://issues.cask.co/browse/CDAP-11421): Shows new pipeline status labels on pipeline run history
- [CDAP-11619](https://issues.cask.co/browse/CDAP-11619): Enables Run & Schedule buttons even when macro values are not filled
- [CDAP-11628](https://issues.cask.co/browse/CDAP-11628): Allows user to disable macros in pipeline runtime arguments

Cherry-picked from #8950.

### CODE REVIEW

**Comment by tonybach** [comment]
The last two commits were actually the first, but I made a mistake in the order when cherry-picking.

File: cdap-ui/app/directives/my-pipeline-configurations/my-batch-pipeline-config/my-batch-pipeline-config.html [hunk 1]
```diff
@@ -264,32 +267,44 @@ <h3 class="modeless-title">
...
+              </button>
+              <span class="num-runtime-args"
+                    ng-if="BatchPipelineConfigCtrl.activeTab === 'runtimeArgs'">
+                  {{ BatchPipelineConfigCtrl.runtimeArguments.pairs.length }}
+                <span ng-if="BatchPipelineConfigCtrl.runtimeArguments.pairs.length === 1">
+                  runtime argument
+                </span>
+                <span ng-if="BatchPipelineConfigCtrl.runtimeArguments.pairs.length > 1">
+                  runtime arguments
+                </span>
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-configurations/my-batch-pipeline-config/my-batch-pipeline-config.html]: see if you can use ngPluralize for this

https://docs.angularjs.org/api/ng/directive/ngPluralize

**Comment by elfenheart** [comment]
👍  LGTM

**Review by elfenheart — APPROVED** [review]
👍  LGTM

**Comment by leacd** [comment]
LGTM

**Comment by tonybach** [comment]
Build: https://builds.cask.co/browse/CDAP-DRC6674-6

**Pull request merged**